### PR TITLE
feat: Add Groq LLM integration

### PR DESCRIPTION
**Pull request by gradenr**
# Description
Adds a LLM integration for Groq models.

Fixes # (issue)

## Type of Change

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

# Suggested Checklist:

- [X] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by marcusschiesser — APPROVED** [review]
Looks great! Thanks @gradenr

File: apps/docs/docs/modules/llms/available_llms/groq.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+const groq = new Groq({
+  apiKey: "<YOUR_API_KEY>",
+});
+
+const serviceContext = serviceContextFromDefaults({ llm: groq });
+```
+
+## Load and index documents
+
+For this example, we will use a single document. In a real-world scenario, you would have multiple documents to index.
```
Comment (marcusschiesser) [inline apps/docs/docs/modules/llms/available_llms/groq.md:17]: @gradenr Great that you added docs. You can even move the examples to the `/examples` folder and reference them in your doc file, see https://github.com/run-llama/LlamaIndexTS/blob/2ec6a529c7d7d9f1fe9aff38f16568bdd3e27072/apps/docs/docs/modules/data_loader.mdx#L5 for how to do that

**Comment by marcusschiesser** [comment]
@gradenr Can you please have a look at the conflict?

File: apps/docs/docs/modules/llms/available_llms/groq.md (hunk 1 above)
Comment (gradenr) [reply apps/docs/docs/modules/llms/available_llms/groq.md]: I moved the "full example" to examples and loaded it with code blocks.  I'm not so sure what to do with the smaller snippets.

**Comment by gradenr** [comment]
> @gradenr Can you please have a look at the conflict?
Rebased

**Review by marcusschiesser — APPROVED** [review]

File: apps/docs/docs/modules/llms/available_llms/groq.md (hunk 1 above)
Comment (marcusschiesser) [reply apps/docs/docs/modules/llms/available_llms/groq.md]: That's fine thanks!

**Pull request merged**